test/e2e/etcdquorumguard_test.go

```diff
@@ -98,23 +98,44 @@ func makeNodeUnSchedulableOrSchedulable(cs *framework.ClientSet, node string, un
 		if err != nil {
 			return err
 		}
-		if n.Spec.Unschedulable != unschedulable {
-			n.Spec.Unschedulable = !unschedulable
-			if _, err := cs.CoreV1Interface.Nodes().Update(n); err != nil {
-				if strings.Contains(err.Error(), "the object has been modified") {
-					fmt.Print("    Node object was modified and not up to date; retrying\n")
-					continue;
-				}
-				return errors.Wrapf(err, "failed to make node %s %sschedulable", node, prefix)
-			}
-		} else {
+		if n.Spec.Unschedulable == unschedulable {
 			fmt.Printf("  Node %s is already %sschedulable", node, prefix)
+			return nil
+		}
+		n.Spec.Unschedulable = unschedulable
+		if _, err := cs.CoreV1Interface.Nodes().Update(n); err != nil {
+			if strings.Contains(err.Error(), "the object has been modified") {
+				fmt.Print("    Node object was modified and not up to date; retrying\n")
+				continue;
+			}
+			return errors.Wrapf(err, "failed to make node %s %sschedulable", node, prefix)
 		}
-		return nil
+		return wait.PollImmediate(1*time.Second, 5*time.Minute, func() (bool, error) {
+			err1 := getMasterNodes(cs)
+			if err1 != nil {
+				fmt.Printf("Error getting master nodes: %s\n", err1.Error())
+				return true, err
```

Comment:
Should this return err1?

Just an optional optimization
```golang
if err := getMasterNodes(cs); err != nil {
   fmt.Printf("Error getting master nodes: %s\n", err.Error())
   return true, err
}
```
that would also allow you to use err vs err1 below.

Reply:
+1